Vagrantfile

```diff
@@ -64,6 +67,9 @@ Vagrant.configure("2") do |config|
   config.vm.define :bootstrap, primary: true do |bootstrap|
     bootstrap.vm.hostname = "bootstrap"
 
+    bootstrap_ip = NETWORK_RANGE + "#{BOOTSTRAP_IP_OFFSET}"
+    bootstrap.vm.network "private_network", ip: bootstrap_ip
```

Comment:
Are we forced to provide an IP address in a private network? Is there no way to let Vagrant allocate IPs, and maybe fill a `hosts` file or similar? It'd be nice not to need to worry about IPAM...

Reply:
No we could use a DHCP of course, but then we'll have to SSH into the VMs to know which IP has been assigned, see https://www.vagrantup.com/docs/networking/private_network.html#dhcp.

One thing that we can do, related to my other comment, is having a yaml file looking as:
```yaml
---
- name: bootstrap
  ip: 192.168.42.10
  role: bootstrap
- name: node1
  ip: 192.168.42.11
  role: workload-plane
...
```
and use this architecture file in the Vagrantfile to setup the VMs.

Reply:
that information is in the provider not in vagrant, `vboxmanage list runningvms | cut -d" " -f 1 | xargs -L 1 -I {} VBoxManage guestproperty get {} '/VirtualBox/GuestInfo/Net/1/V4/IP'` :stuck_out_tongue:

Reply:
Yep we could find it in Virtualbox if need to, but I don't think there is a real difference between finding it out this way and having static IPs. Anyway, that's not for now.